packages/@dcos/sorting/index.ts

```diff
@@ -0,0 +1,61 @@
+export enum ComparisonResult {
+  BIGGER = 1,
+  EQUAL = 0,
+  SMALLER = -1
+}
+export type Comparator<T> = (a: T, b: T) => ComparisonResult;
+export type Lense<T> = (data: T) => any;
+
+export function sort<T>(comps: Array<Comparator<T>>, data: T[]): T[] {
+  return [...data].sort((a, b) => {
+    for (const comperator of comps) {
+      const result = comperator(a, b);
+
+      if (result !== 0) {
+        return result;
+      }
+    }
+
+    return 0;
```

Comment:
Why not default to the javascript default sort behavior? (which I believe is the UNICODE codepoint value of T converted to string).

Reply:
My reasoning was if I sort by nothing I expect no change.

Reply:
I think my expectation would actually be that I can call `sort([1,4,5])` without any comparator. but maybe that does not make sense on this library.

Reply:
What about this point?

Reply:
Good point, I would like to note this down as a trade-off.

Reply:
Alright.